release.py

```diff
@@ -0,0 +1,848 @@
+
+    while True:
+        user_input = input(
+            f'Input tag to release (press enter to use the suggested tag '
+            f"{suggested_tag or 'required'}), or 'c' to cancel: "
+        ).strip()
+
+        if user_input.lower() == 'c':
+            logger.warning('Release canceled.')
+            exit(0)
+
+        new_tag = user_input if user_input else suggested_tag
+
+        if not new_tag:
+            logger.error('Error: No tag specified and no suggestion available')
+            continue
+
+        if not re.match(r'^\d+\.\d+\.\d+$', new_tag):
+            logger.error('Error: Tag must be in format X.Y.Z')
+            continue
+
+        release_page = f'https://github.com/{owner}/{repo.name}/releases'
+        logger.warning(f'Check out the releases page: {release_page} before confirming!')
+
+        confirm = (
+            input(f"Confirm creating tag '{new_tag}' on branch '{branch_name}'? [y/N]: ")
+            .strip()
+            .lower()
+        )
+
+        if confirm == 'y':
+            break
+
+        logger.info("Let's try again...")
+
+    return new_tag
+
+
+def create_draft_release(
+    repo: github.Repository.Repository, tag: str, branch: str
+) -> github.GitRelease.GitRelease | None:
+    """Create a draft release with auto-generated notes."""
+    max_retries = 3
+    for attempt in range(1, max_retries + 1):
+        try:
+            release = repo.create_git_release(
+                tag=tag,
+                name=tag,
+                draft=True,
+                generate_release_notes=True,
+                target_commitish=branch,
+            )
+            return release
+        except Exception as e:
+            logger.error(f'Attempt {attempt} failed to create release: {e}')
+        if attempt == max_retries:
+            raise
+        time.sleep(1)
+
+    return None
+
+
+def parse_release_notes(release_notes: str) -> tuple[dict[str, list[tuple[str, str]]], str | None]:
+    """Parse auto-generated release notes into categories.
+
+    The "New Contributors" section is removed.
+    The PRs are categorized into 'feat', 'fix', 'docs', 'test', and 'ci'.
+    The full changelog line is returned separately.
+
+    Returns:
+        A tuple containing:
+        - A dict with conventional commit types as keys and lists of tuples (description, PR link)
+          as values.
+        - The full changelog line if present, or None if not found.
+    """
+    release_notes = re.sub(
+        r'(## New Contributors.*?)(\n|$)', r'\2', release_notes, flags=re.DOTALL
+    )
+    categories: dict[str, list[tuple[str, str]]] = {
+        'feat': [],
+        'fix': [],
+        'docs': [],
+        'test': [],
+        'ci': [],
+    }
+    full_changelog_line = None
+
+    for line in release_notes.splitlines():
+        line = line.strip()
+        if line.startswith('* '):
+            match = re.match(r'\* (\w+): (.*) by @\w+ in (.*)', line)
```

Comment:
I think we want to keep (in the release notes, not so much the changelog) acknowledgement of external contributions, don't we?

Reply:
Per [discussion here](https://docs.google.com/document/d/1fkoqp1A2SMiOnwHubOJVRkuD1GpvcYxSDmfExYIZO88/edit?tab=t.0) (internal link), we have agreed on "Drop the GH usernames (even if they come from outside the team)."